receiver/telemetrygeneratorreceiver/logs_receiver.go

```diff
@@ -0,0 +1,60 @@
+// Copyright observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package telemetrygeneratorreceiver //import "github.com/observiq/bindplane-agent/receiver/telemetrygeneratorreceiver"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer" // newLogsReceiver creates a new logs specific receiver.
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.uber.org/zap"
+)
+
+type logsGeneratorReceiver struct {
+	telemetryGeneratorReceiver
+	nextConsumer consumer.Logs
+	generators   []generator
+}
+
+func newLogsReceiver(ctx context.Context, logger *zap.Logger, cfg *Config, nextConsumer consumer.Logs) (*logsGeneratorReceiver, error) {
+	lr := &logsGeneratorReceiver{
+		nextConsumer: nextConsumer,
+	}
+	r, err := newTelemetryGeneratorReceiver(ctx, logger, cfg, lr)
+	if err != nil {
+		return nil, err
+	}
+
+	lr.telemetryGeneratorReceiver = r
+	r.supportedTelemetry = component.DataTypeLogs
+
+	lr.generators = newGenerators(cfg, logger, r.supportedTelemetry)
+
+	return lr, nil
+}
+
+// generateTelemetry
+func (r *logsGeneratorReceiver) produce() error {
+	logs := plog.NewLogs()
+	for _, g := range r.generators {
+		l := g.GenerateLogs()
```

Comment:
Should do a check here with `g.SupportsType(component.Logs)`, and skip if it's false.

Reply:
newGenerators only returns generators that support the telemetry type